### ISSUE TITLE
Add .Net 6 support

### ISSUE THREAD

**Issue by deMD** [HangfireIO/Hangfire#2086]
We are using a .Net 6 app with hangfire but run into the issue that we can't use the default .net serviceprovider in .Net 6

It seems we are missing the public static IServiceCollection AddHangfireServer([NotNull] this IServiceCollection services, [NotNull] Action<BackgroundJobServerOptions> optionsAction) method for .Net containers in the .Net 6 as referenced here as it is hidden behind preprocessor directive that don't include the latest .Net versions: https://github.com/HangfireIO/Hangfire/blob/master/src/Hangfire.AspNetCore/HangfireServiceCollectionExtensions.cs#L107

**Comment by morrislaw715** [comment]
I'm also using .net 6 to build a hangfire server just few line coding.
I put below code in program.cs.

```
using Hangfire;

var builder = WebApplication.CreateBuilder(args);
var sqlConBuilder = new SqlConnectionStringBuilder();
sqlConBuilder.ConnectionString = builder.Configuration.GetConnectionString("HangfireConnection");
string strConstr = sqlConBuilder.ConnectionString;

builder.Services.AddHangfire(options =>
{
    options.UseSqlServerStorage(strConstr);
});
builder.Services.AddHangfireServer(options =>
{
    options.SchedulePollingInterval = TimeSpan.FromSeconds(1);
});
var app = builder.Build();

app.MapGet("/", () => "This is Hangfire server!");
app.UseHangfireDashboard();
app.MapHangfireDashboard();

app.Run();
```

**Comment by deMD** [comment]
> I'm also using .net 6 to build a hangfire server just few line coding. I put below code in program.cs.
> 
> ```
> using Hangfire;
> 
> var builder = WebApplication.CreateBuilder(args);
> var sqlConBuilder = new SqlConnectionStringBuilder();
> sqlConBuilder.ConnectionString = builder.Configuration.GetConnectionString("HangfireConnection");
> string strConstr = sqlConBuilder.ConnectionString;
> 
> builder.Services.AddHangfire(options =>
> {
>     options.UseSqlServerStorage(strConstr);
> });
> builder.Services.AddHangfireServer(options =>
> {
>     options.SchedulePollingInterval = TimeSpan.FromSeconds(1);
> });
> var app = builder.Build();
> 
> app.MapGet("/", () => "This is Hangfire server!");
> app.UseHangfireDashboard();
> app.MapHangfireDashboard();
> 
> app.Run();
> ```

Yeah it seems to work as long as you don't have dependcies in your jobs. We have a job where we want to index data in ElasticSearch, but the ElasticClient that is registered in DI is always null when the job is started.

**Comment by aviatrix** [comment]
https://docs.hangfire.io/en/latest/background-methods/using-ioc-containers.html

**Comment by tyaramis** [comment]
Here it is an example to configure it in a .Net core 6.0 Windows Service. 
> 

```
IHost host = Host.CreateDefaultBuilder(args)
    .UseWindowsService(options =>
    {
        options.ServiceName = "App Backend Service";
    })
    .ConfigureLogging(
        (ctx, logging) =>
        {
            if (ctx.HostingEnvironment.IsDevelopment() || args.Contains("--verbose"))
            {
                logging.AddConsole();
            }
            else
            {
                logging.ClearProviders();
                logging.SetMinimumLevel(LogLevel.Debug);
            }
        })
    .UseNLog()
    .ConfigureServices(
    (hostContext, services) =>
    {
        services.AddHangfire(configuration =>
            configuration.UseSqlServerStorage(hostContext.Configuration.GetConnectionString("HangfireConnection")));

        services.AddHangfireServer(options =>
        {
            options.SchedulePollingInterval = TimeSpan.FromSeconds(1);
        });

        services.AddDbContextFactory<AppDbContext>(opt => opt.UseSqlServer(hostContext.Configuration.GetConnectionString("AppDbConnection")));        

        services.AddScoped<IHangfireAccessor, HangfireAccessor>();
        services.AddScoped<IChangeDetectionService, ChangeDetectionService>();
		
		services.AddHostedService<Worker>();
    })
    .ConfigureLogging((context, logging) =>
    {
        logging.AddConfiguration(context.Configuration.GetSection("Logging"));
    })
    .Build();

await host.RunAsync();


```

**Comment by edusoccer1121** [comment]
Why the double ConfigureLogging customization ? I think you are overriding config.

> Here it is an example to configure it in a .Net core 6.0 Windows Service.
> 
> > 
> 
> ```
> IHost host = Host.CreateDefaultBuilder(args)
>     .UseWindowsService(options =>
>     {
>         options.ServiceName = "App Backend Service";
>     })
>     .ConfigureLogging(
>         (ctx, logging) =>
>         {
>             if (ctx.HostingEnvironment.IsDevelopment() || args.Contains("--verbose"))
>             {
>                 logging.AddConsole();
>             }
>             else
>             {
>                 logging.ClearProviders();
>                 logging.SetMinimumLevel(LogLevel.Debug);
>             }
>         })
>     .UseNLog()
>     .ConfigureServices(
>     (hostContext, services) =>
>     {
>         services.AddHangfire(configuration =>
>             configuration.UseSqlServerStorage(hostContext.Configuration.GetConnectionString("HangfireConnection")));
> 
>         services.AddHangfireServer(options =>
>         {
>             options.SchedulePollingInterval = TimeSpan.FromSeconds(1);
>         });
> 
>         services.AddDbContextFactory<AppDbContext>(opt => opt.UseSqlServer(hostContext.Configuration.GetConnectionString("AppDbConnection")));        
> 
>         services.AddScoped<IHangfireAccessor, HangfireAccessor>();
>         services.AddScoped<IChangeDetectionService, ChangeDetectionService>();
> 		
> 		services.AddHostedService<Worker>();
>     })
>     .ConfigureLogging((context, logging) =>
>     {
>         logging.AddConfiguration(context.Configuration.GetSection("Logging"));
>     })
>     .Build();
> 
> await host.RunAsync();
> ```
